pkg/runtime/framework/plugins/torch/torch.go

```
@@ -188,3 +231,27 @@ func calculateNumProcPerNode(
 	}
 	return intstr.FromInt32(defaultCPU), false
 }
+
+// getRecipeAndConfig returns the recipe and config file name based on the number of nodes,
+// number of processes per node, runtime reference name, and command line arguments.
+func getRecipeAndConfig(numNodes int32, numProcPerNode intstr.IntOrString, runtimeRefName string, _ []string) (string, string) {
+	recipe := constants.TorchTuneFullFinetuneDistributed
+	suffix := constants.TorchTuneFullFinetuneMultiDevicesConfigSuffix
+	if numNodes == 1 && numProcPerNode.Type == intstr.Int && numProcPerNode.IntVal == 1 {
+		recipe = constants.TorchTuneFullFinetuneSingleDevice
+		suffix = constants.TorchTuneFullFinetuneSingleDeviceConfigSuffix
+	} else if numNodes > 1 {
+		suffix = constants.TorchTuneFullFinetuneMultiNodesConfigSuffix
+	}
+
+	return recipe, fmt.Sprintf("%s%s.yaml", getModelFromRuntimeRef(runtimeRefName), suffix)
+}
+
+func getModelFromRuntimeRef(runtimeRefName string) string {
+	fields := strings.Split(runtimeRefName, "-")
+	if len(fields) != 3 {
```

Shall we return an error if runtime ref has incorrect name ?
I think, we should validate it in the Webhook.

---

From my perspective, this function is designed only for CTR related to `torchtune`, which means that we'll only have predefined `runtimeRefName` argument passed to this function. So, maybe it's unnecessary to validate it? WDYT @andreyvelich

---

That's correct, but if runtime name is set incorrectly for the `torchtune` the orchestration will fail, right ?
Thus, I suggest that we throw an error here to user rather then return empty string.
If can of empty string users will get error when the `tune run` command is executed.

---

SGTM. I'll add a validation in `Validate` API soon:)

---

Emmm... It seems that I've already verified it in the webhook:

https://github.com/kubeflow/trainer/blob/87f68ff453f815baceb387b8da4d5417ce6da329/pkg/runtime/framework/plugins/torch/torch.go#L90-L95

If this function returns empty string, it will failed to pass the check of `TorchTuneSupportedPretrainedModels`

---

Alright, sound good.